src/mscorlib/src/Microsoft/Win32/Win32Native.cs

```diff
@@ -687,7 +687,15 @@ internal class WIN32_FIND_DATA
         internal static extern bool SetEnvironmentVariable(string lpName, string lpValue);
 
         [DllImport(Interop.Libraries.Kernel32, CharSet = CharSet.Auto, SetLastError = true, BestFitMapping = false)]
-        internal static extern int GetEnvironmentVariable(string lpName, [Out]StringBuilder lpValue, int size);
+        private static extern unsafe int GetEnvironmentVariable(string lpName, char* lpValue, int size);
```

Comment:
you've also eliminated the StringBuilder marshaling it was doing before, which forced a copy.
https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-pinvokes.md

Reply:
Yup :)